### PR TITLE
feat: drop youtube_dl crate

### LINKED ISSUE #102: Drop `youtube_dl` crate in favour of `yt-dlp` command exec
We're only using the [`youtube_dl`](https://crates.io/crates/youtube_dl) crate for fetching the songs within a playlist.
This is overkill since we're using yt-dlp as well. Investigate executing a yt-dlp command (possibly with the `--flat-playlist` flag).

### PR DESCRIPTION
**Pull request by aquelemiguel**
Closes #102.

* Drops `youtube_dl` crate that was being used when enqueueing a playlist.
* Removes other crates that were being used in the lyrics command.

### CODE REVIEW

**Review by afonsojramos — COMMENTED** [review]
Looks good code wise! Goodbye useless dependencies 👋 

Will need to test with some playlists before approving.

File: src/commands/play.rs [hunk 1]
```diff
@@ -145,16 +150,19 @@ async fn calculate_time_until_play(queue: &[TrackHandle], flag: &PlayFlag) -> Op
...
+        let reader = BufReader::new(stdout);
+        let lines: Vec<String> = reader.lines().flatten().collect();
 
-        for entry in entries.iter() {
-            let url = format!(
-                "https://www.youtube.com/watch?v={}",
-                entry.url.as_ref().unwrap()
-            );
+        for line in lines.iter() {
+            let entry: Value = serde_json::from_str(line).unwrap();
```
Comment (joao-conde) [inline src/commands/play.rs:164]: Can't we get away with implicit typing here (removing the `: Value`) ?

Other than that looks good but I haven't been able to test it

Comment (aquelemiguel) [reply src/commands/play.rs:164]: Doesn't look like it, type annotations are needed in the line below.

**Review by afonsojramos — APPROVED** [review]
Merged with `main` so that `Cargo.lock` was updated.

**Pull request merged**